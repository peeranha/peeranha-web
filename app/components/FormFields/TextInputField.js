import React from 'react';
import PropTypes from 'prop-types';
import { FormattedMessage } from 'react-intl';

function TextInputField({
  input,
  label,
  readOnly,
  disabled,
  meta: { touched, error, warning },
}) {
  return (
    <div>
      <h6>{label}</h6>
      <input
        {...input}
        disabled={disabled}
        readOnly={readOnly}
        placeholder={label || ''}
        type="text"
        className="form-control"
      />
      <h6 className="text-danger">
        {touched &&
          ((error && <FormattedMessage {...error} />) ||
            (warning && <FormattedMessage {...warning} />))}
      </h6>
    </div>
  );
}

TextInputField.propTypes = {
  input: PropTypes.object,
  meta: PropTypes.object,
  disabled: PropTypes.bool,
  readOnly: PropTypes.bool,
  label: PropTypes.string,
};

export default TextInputField;
