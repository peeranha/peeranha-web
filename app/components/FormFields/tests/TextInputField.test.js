import { TextInputField } from '../TextInputField';

const props = {
  input: {},
  label: 'label',
  readOnly: false,
  disabled: false,
  meta: {},
  placeholder: 'placeholder',
  isSearchable: false,
  tip: 'tip',
};

describe('TextInputField test', () => {
  it('test', () => {
    expect(TextInputField(props)).toMatchSnapshot();
  });
});
