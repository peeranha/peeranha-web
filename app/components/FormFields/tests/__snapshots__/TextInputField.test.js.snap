// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInputField test test by snapshots default 1`] = `
.c1 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-transition: 0.5s;
  transition: 0.5s;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.c1 .chevron {
  position: absolute;
  right: 14px;
  cursor: pointer;
  color: #bdbdbd;
}

.c1 input {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  height: 40px;
  border-radius: 3px;
  border: 1px solid #bdbdbd;
  box-shadow: 0 0 0 3px rgba(0,0,0,0);
  padding: 9px 42px 9px 14px;
  color: #282828;
  font-family: Source Sans Pro,sans-serif;
  font-size: 16px;
  line-height: 20px;
  outline: none;
}

.c1 input:focus {
  box-shadow: 0 0 0 3px #576fed66;
  border-color: #576fed;
}

.c1 input:disabled {
  opacity: 0.5;
}

.c0 {
  color: #282828;
  opacity: 1;
  font-weight: normal;
  font-size: 16px;
  line-height: 20px;
  font-style: normal;
  font-family: Source Sans Pro,sans-serif;
  text-align: left;
  margin-bottom: 8px;
  font-size: 16px;
  font-weight: 600;
}

.c0 [data-icon='icon'] {
  stroke: #7b7b7b;
  fill: #7b7b7b;
}

<div
  className="mb-2"
>
  <h6
    className="c0"
  >
    string
  </h6>
  <div
    className="row align-items-start"
  >
    <div
      className="col-xl-6 mb-1 c1"
    >
      <input
        type="text"
      />
    </div>
  </div>
</div>
`;
